src/pages/registration/resource_type_tab/sub_type_forms/artistic_types/music_performance/ConcertModal.tsx

```diff
@@ -118,43 +138,81 @@ export const ConcertModal = ({ concert, onSubmit, open, closeModal }: ConcertMod
   const [removeWorkItemIndex, setRemoveWorkItemIndex] = useState(-1);
   const closeConfirmDialog = () => setRemoveWorkItemIndex(-1);
 
+  const [partOfSeries, setPartOfSeries] = useState(concert?.concertSeries ? true : false);
+
+  const validateForm = (values?: Concert): FormikErrors<Concert> => {
+    if (!values) {
+      return {};
+    }
+
+    try {
+      validateYupSchema<Concert>(values, validationSchema, true, {
+        partOfSeries: partOfSeries,
+      });
+    } catch (err) {
+      return yupToFormErrors(err);
+    }
+    return {};
+  };
+
   return (
     <Dialog open={open} onClose={closeModal} maxWidth="md" fullWidth>
       <DialogTitle>
         {concert
           ? t('registration.resource_type.artistic.edit_concert')
           : t('registration.resource_type.artistic.add_concert')}
       </DialogTitle>
+
       <Formik
         initialValues={concert ?? emptyConcert}
-        validationSchema={validationSchema}
+        validate={validateForm}
+        initialErrors={validateForm(concert)}
         onSubmit={(values) => {
           onSubmit(values);
           closeModal();
         }}>
         {({ values, errors, touched, isSubmitting, setFieldValue, setFieldTouched }: FormikProps<Concert>) => (
           <Form noValidate>
             <DialogContent sx={{ display: 'flex', flexDirection: 'column', gap: '1rem' }}>
-              <Field name="partOfSeries">
-                {({ field }: FieldProps<boolean>) => (
-                  <FormControlLabel
-                    data-testid={dataTestId.registrationWizard.resourceType.concertPartOfSeries}
-                    label={t('registration.resource_type.artistic.concert_part_of_series')}
-                    control={
-                      <Checkbox
-                        checked={field.value}
+              <FormControlLabel
+                data-testid={dataTestId.registrationWizard.resourceType.concertSeriesCheckbox}
+                label={t('registration.resource_type.artistic.concert_part_of_series')}
+                control={
+                  <Checkbox
+                    checked={partOfSeries}
+                    onChange={() => {
+                      if (!partOfSeries) {
+                        setFieldValue('time', emptyPeriod);
+                        setFieldValue('concertSeries', '');
+                        setPartOfSeries(!partOfSeries);
+                      } else {
+                        setFieldValue('time', emptyInstant);
+                        setFieldValue('concertSeries', undefined);
+                        setPartOfSeries(!partOfSeries);
+                      }
```

Comment:
Dette er kanskje en forenkling som gjør samme nytte?
```suggestion
                      if (!partOfSeries) {
                        setFieldValue('time', emptyPeriod);
                      } else {
                        setFieldValue('time', emptyInstant);
                        setFieldValue('concertSeries', '');
                      }
                      setPartOfSeries(!partOfSeries);
```